src/main/java/lotto/view/InputView.java

```diff
@@ -0,0 +1,74 @@
+package lotto.view;
+
+import camp.nextstep.edu.missionutils.Console;
+import java.util.Arrays;
+import java.util.List;
+import lotto.app.dto.WinningNumberRequestDto;
+import lotto.domain.PositiveNumber;
+import lotto.exception.ErrorMessage;
+
+public class InputView {
+
+    private static final String DELIMITER = ",";
+
+    private final InputValidator inputValidator;
+
+    public InputView(InputValidator inputValidator) {
+        this.inputValidator = inputValidator;
+    }
+
+    public PositiveNumber getPrice() {
+        while (true) {
+            try {
+                System.out.println("구입금액을 입력해 주세요.");
+                long price = Long.parseLong(Console.readLine());
+                inputValidator.validateDivisibleByThousand(price);
+                return new PositiveNumber(price);
+            } catch (NumberFormatException e) {
+                System.out.println(ErrorMessage.INVALID_DATA_TYPE.get());
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
```

Comment:
Supplier 함수형 인터페이스를 사용하면 반복되는 while-try-catch문을 제거해볼 수 있겠습니다

```suggestion
    public PositiveNumber getPrice() {
        return inputWithRetry(() -> {
                System.out.println("구입금액을 입력해 주세요.");
                long price = Long.parseLong(Console.readLine());
                inputValidator.validateDivisibleByThousand(price);
                return new PositiveNumber(price);
        });
    }
    
    private <T> T inputWithRetry(Supplier<T> supplier) {
        while (true) {
            try {
                return supplier.get();
            } catch (NumberFormatException e) {
                System.out.println(ErrorMessage.INVALID_DATA_TYPE.get());
            } catch (IllegalArgumentException e) {
                System.out.println(e.getMessage());
            }
        }
    }
```